change-validation/src/main/scala/com/normation/plugins/changevalidation/NotificationService.scala

```diff
@@ -0,0 +1,237 @@
+      mailParameter <- state match {
+        case Validation => getValidationMailParameter(path)
+        case Deployment => getDeploymentMailParameter(path)
+      }
+//      content <- getContentFromTemplate(path)
+      sendingMail <- sendEmail(serverConfig, "toto", mailParameter)
+    } yield (sendingMail)
+
+    notif.catchAll {
+      err =>
+        for {
+          _ <- logger.error(s"Sending `${state.id.value}` notification have failed, cause by : ${err}")
+        } yield {
+          state
+        }
+    }
+  }
+
+  //TO-DO : handle errors with ZIO, exceptions, Task ?
+  def sendEmail(conf: SMTPConf, content: String, mailParameter: EmailInfo) = {
+
+    val prop = new Properties()
+    prop.put("mail.smtp.host", conf.smtpHostServer)
+    prop.put("mail.smtp.port", conf.port)
+    prop.put("mail.smtp.starttls.enable", "true")
+
+    val session = (conf.login, conf.password) match {
+      case (Some(l), Some(p)) =>
+        prop.put("mail.smtp.auth", "true");
+        val auth = new Authenticator() {
+          override protected def getPasswordAuthentication = new PasswordAuthentication(l.value, p)
+        }
+        Session.getInstance(prop, auth)
+      case (_, None) =>
+        prop.put("mail.smtp.auth", "false");
+        Session.getInstance(prop, null)
+      case (None, _) =>
+        prop.put("mail.smtp.auth", "false");
+        Session.getInstance(prop, null)
+    }
+    try {
+      val message = new MimeMessage(session);
+      message.setFrom(new InternetAddress(conf.email.value))
+      message.setRecipients(
+        Message.RecipientType.TO,
+        mailParameter.to.map(_.value).mkString(",")
+      )
+      val reply: Array[Address] = mailParameter.replyTo.map(e => new InternetAddress(e.value)).toArray
+      message.addRecipients(Message.RecipientType.BCC, mailParameter.bcc.map(_.value).mkString(","))
+      message.addRecipients(Message.RecipientType.CC, mailParameter.cc.map(_.value).mkString(","))
+      message.setReplyTo(reply)
+      message.setSubject(mailParameter.subject);
+
+      message.setText(content);
+      Transport.send(message)
+      mailParameter.succeed
+    } catch {
+      case e: MessagingException =>
+        logger.error(s"Sending mail to notification have failed, cause by : ${e.getMessage}").fail
+    }
+  }
+
+  def getConfig(path: String): Config = {
+    val file = new File(path)
+    val configResource = if(file.exists && file.canRead) {
+      Some(FileSystemResource(file))
+    } else {
+      logger.error("Can not find configuration file specified by JVM property %s: %s ; abort")
+      throw new Exception("Configuration file not found: %s".format(file.getPath))
+    }
+    ConfigFactory.load(ConfigFactory.parseFile(configResource.value.file))
+  }
+
+
+  //TO-DO have to deal with the case where a parameter is not set (logger)
+  def getSMTPConf(path: String): IO[Throwable, SMTPConf] = {
+    for {
+      config     <- ZIO.fromTry(Try(getConfig(path)))
```

Comment:
That should be 

IOResult.effect("An error occurs while parsing SMTP conf in ${path} {
 val hostServer = config.getString("hostServer")
 ...
  val login = {
     val l = config.getString("login")
     if (l.isEmpty) {None} else {Some(Username(l))}
  }
  SMTPConf(hostServer .... ) 
}


}